From 8615ee27f5a86e1c8c1161f3b58ece007019e86f Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Wed, 29 Nov 2017 18:07:39 +0200
Subject: [PATCH 2132/2241] net: pp2x: fix: allocate tx descriptors zeroed

Reserved/unused fields in tx-descriptors must be ZERO.
Driver doesn't clear them in run-time to improve performance.
The descriptors are not static-variable but run-time allocated,
so the allocator must Z-alloc procedure.

The problem could be easy emulated on original code
by setting the allocated area with 0xA5 -- driver doesn't work.

Change-Id: I1971cd75151f9931f022e7a57fd7a2a09c0b1dba
Signed-off-by: Yan Markman <ymarkman@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/47064
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-by: Yuval Caduri <cyuval@marvell.com>
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
---
 drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
index 1c15d20..2a75614 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
@@ -1209,7 +1209,7 @@ static int mv_pp2x_aggr_txq_init(struct platform_device *pdev,
 	dma_addr_t first_desc_phy;
 
 	/* Allocate memory for TX descriptors, ensure it can be 512B aligned. */
-	aggr_txq->desc_mem = dma_alloc_coherent(&pdev->dev,
+	aggr_txq->desc_mem = dma_zalloc_coherent(&pdev->dev,
 		MVPP2_DESCQ_MEM_SIZE(desc_num),
 		&aggr_txq->descs_phys, GFP_KERNEL);
 	if (!aggr_txq->desc_mem)
-- 
2.7.4

